apps/web/src/components/Composer/Editor/index.tsx

```diff
@@ -18,6 +21,20 @@ const TRANSFORMERS = [...TEXT_FORMAT_TRANSFORMERS];
 
 const Editor: FC = () => {
   const setPublicationContent = usePublicationStore((state) => state.setPublicationContent);
+  const attachments = usePublicationStore((state) => state.attachments);
+  const addAttachments = usePublicationStore((state) => state.addAttachments);
+
+  const handlePaste = async (files: FileList) => {
+    if (attachments.length === 4 || attachments.length + files.length > 4) {
+      return toast.error('You can only upload 4 files.');
+    }
+    if (files) {
+      const newAttachments = await uploadToIPFS(files);
+      if (newAttachments.length) {
+        addAttachments([...newAttachments]);
```

Comment:
```suggestion
        addAttachments([...newAttachments]);
```
Just wondering why we have to spread and again send it as an array instead of sending newAttachments directly?